src/polymer/polymer-element.ts

```diff
@@ -334,52 +334,64 @@ function inheritFrom(element: PolymerElement, superElement: PolymerExtension) {
   // TODO(justinfagnani): fixup and use inheritValues, but it has slightly odd
   // semantics currently
 
-  for (const superProperty of superElement.properties) {
-    const newProperty = Object.assign({}, superProperty);
-    if (!newProperty.inheritedFrom) {
-      const superName = getSuperName(superElement);
-      if (superName) {
-        newProperty.inheritedFrom = superName;
-      }
-    }
-    element.properties.push(newProperty);
-  }
+  const superName = getSuperName(superElement);
+  _inheritVals(
+      element.properties, superElement.properties, superName, element.warnings);
+  _inheritVals(
+      element.methods, superElement.methods, superName, element.warnings);
+  _inheritVals(
+      element.attributes, superElement.attributes, superName, element.warnings);
+  _inheritVals(
+      element.events, superElement.events, superName, element.warnings);
 
-  for (const superMethod of superElement.methods) {
-    const newMethod = Object.assign({}, superMethod);
-    if (!newMethod.inheritedFrom) {
-      const superName = getSuperName(superElement);
-      if (superName) {
-        newMethod.inheritedFrom = superName;
-      }
-    }
-    element.methods.push(newMethod);
-  }
+  // TODO(justinfagnani): slots, listeners, observers, dom-module?
+  // What actually inherits?
+}
 
-  for (const superAttribute of superElement.attributes) {
-    const newAttribute = Object.assign({}, superAttribute);
-    if (!newAttribute.inheritedFrom) {
-      const superName = getSuperName(superElement);
-      if (superName) {
-        newAttribute.inheritedFrom = superName;
+type PropLike = {
+  name: string,
+  sourceRange?: SourceRange,
+  inheritedFrom?: string,
+  privacy?: Privacy
+};
+/**
+ * Mutates `existing`
+ *
+ * TODO(rictic): unify this with inheritValues in this file.
+ */
+function _inheritVals<P extends PropLike>(
+    existing: P[],
+    superVals: P[],
+    superName: string | undefined,
+    warnings: Warning[]) {
+  const existingVals = new Map(existing.map((e) => [e.name, e] as [string, P]));
+  for (const superVal of superVals) {
+    if (existingVals.has(superVal.name)) {
+      if (superVal.privacy === 'private') {
```

Comment:
I had a different response here where I was arguing both sides, but ended up convincing myself that this warning should be on by default :)

Basically a private field ideally shouldn't induce a warning on a subclass since it shouldn't be visible, but JavaScript's not ideal, and we should warn when this situation does occur to avoid mistakes. Someone can turn it off it necessary.